driver-sync/src/test/functional/com/mongodb/client/unified/UnifiedCrudHelper.java

```diff
@@ -226,7 +229,8 @@ OperationResult executeFind(final BsonDocument operation) {
     OperationResult createFindCursor(final BsonDocument operation) {
         FindIterable<BsonDocument> iterable = createFindIterable(operation);
         return resultOf(() -> {
-            entities.addCursor(operation.getString("saveResultAsEntity").getValue(), iterable.cursor());
+            entities.addCursor(operation.getString("saveResultAsEntity", new BsonString(createRandomEntityId())).getValue(),
```

Comment:
The advantage of storing the cursor in `entities` with a random identifier is that it will ensure that the cursor is closed at the end of the test.